File: src/test/java/br/campos/poi/POIServiceTests.java
package br.campos.poi;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit.jupiter.SpringExtension;

import br.campos.poi.POIIncApplication;
import br.campos.poi.model.transfer.POIDTO;
import br.campos.poi.service.abst.IPOIService;
import br.campos.poi.util.POIException;

/**
 * Testes para o componente IPOIService e POIService
 * 
 * @author Caio
 */
@AutoConfigureMockMvc
@ExtendWith(SpringExtension.class)
@SpringBootTest(classes = POIIncApplication.class)
public class POIServiceTests {
	@Autowired
	IPOIService poiService;

	// Testes em create
	@Test
	public void create() throws POIException {
		final Double mOne = Double.valueOf(-1);
		final Double dValue = Double.valueOf(1435674.003);
		assertThrows(POIException.class, () -> {
			poiService.create(null);
		});
		assertThrows(POIException.class, () -> {
			POIDTO dto = new POIDTO(null, null, null);
			poiService.create(dto);
		});
		assertThrows(POIException.class, () -> {
			POIDTO dto = new POIDTO(null, mOne, null);
			poiService.create(dto);
		});
		assertThrows(POIException.class, () -> {
			POIDTO dto = new POIDTO(null, null, mOne);
			poiService.create(dto);
		});
		assertThrows(POIException.class, () -> {
			POIDTO dto = new POIDTO("A", null, null);
			poiService.create(dto);
		});
		assertThrows(POIException.class, () -> {
			POIDTO dto = new POIDTO("B", mOne, null);
			poiService.create(dto);
		});
		assertThrows(POIException.class, () -> {
			POIDTO dto = new POIDTO("C", null, mOne);
			poiService.create(dto);
		});
		assertThrows(POIException.class, () -> {
			POIDTO dto = new POIDTO("D", mOne, dValue);
			poiService.create(dto);
		});
		assertThrows(POIException.class, () -> {
			POIDTO dto = new POIDTO("E", dValue, mOne);
			poiService.create(dto);
		});
		assertThrows(POIException.class, () -> {
			POIDTO dto = new POIDTO("", dValue, dValue);
			poiService.create(dto);
		});
		// Desabilitado para base em atividade
		/*
		 * POIDTO dtoO = new POIDTO("F", dValue, dValue);
		 * assertNotNull(poiService.create(dtoO)); assertThrows(POIException.class, () ->
		 * { POIDTO dto = new POIDTO("F", dValue, dValue); poiService.create(dto); });
		 * dtoO = new POIDTO("G", 0D, 10D); assertNotNull(poiService.create(dtoO)); dtoO
		 * = new POIDTO("H", 10D, 0D); assertNotNull(poiService.create(dtoO));
		 */
	}

	// Testes em update
	@Test
	public void update() throws POIException {
		final Double mOne = Double.valueOf(-1);
		final Double dValue = Double.valueOf(1435674.003);
		assertThrows(POIException.class, () -> {
			poiService.update(null);
		});
		assertThrows(POIException.class, () -> {
			POIDTO dto = new POIDTO(null, null, null);
			poiService.update(dto);
		});
		assertThrows(POIException.class, () -> {
			POIDTO dto = new POIDTO("", dValue, dValue);
			poiService.update(dto);
		});
		assertThrows(POIException.class, () -> {
			POIDTO dto = new POIDTO("F", mOne, null);
			poiService.update(dto);
		});
		assertThrows(POIException.class, () -> {
			POIDTO dto = new POIDTO("F", null, mOne);
			poiService.update(dto);
		});
		assertThrows(POIException.class, () -> {
			POIDTO dto = new POIDTO("F", mOne, dValue);
			poiService.update(dto);
		});
		assertThrows(POIException.class, () -> {
			POIDTO dto = new POIDTO("F", dValue, mOne);
			poiService.update(dto);
		});
		// Desabilitado para base em atividade
		/*
		 * assertThrows(POIException.class, () -> { POIDTO dto = new POIDTO("A", dValue,
		 * dValue); poiService.update(dto); }); POIDTO dtoO = new POIDTO("F", 1D, 10D);
		 * assertNotNull(poiService.update(dtoO));
		 */
	}

	// Testes em find e findDTO
	@Test
	public void findDTO() throws POIException {
		assertThrows(POIException.class, () -> {
			poiService.findDTO(null);
		});
		assertThrows(POIException.class, () -> {
			POIDTO dto = new POIDTO(null, null, null);
			poiService.findDTO(dto);
		});
		assertThrows(POIException.class, () -> {
			POIDTO dto = new POIDTO("", null, null);
			poiService.findDTO(dto);
		});
		// Desabilitado para base em atividade
		/*
		 * assertThrows(POIException.class, () -> { POIDTO dto = new POIDTO("A", null,
		 * null); poiService.findDTO(dto); }); POIDTO dtoO = new POIDTO("H", null,
		 * null); assertNotEquals(poiService.findDTO(dtoO).toString(),
		 * "POIDTO(nome=H, x=10, y=0)");
		 * 
		 * assertTrue(poiService.findDTO(0D, 0D, 0D).isEmpty());
		 * assertTrue(poiService.findDTO(0D, 0D, 10D).size() == 2);
		 * 
		 * assertTrue(poiService.findDTO().size() == 3);
		 */
	}

	// Testes em delete
	@Test
	public void delete() throws POIException {
		assertThrows(POIException.class, () -> {
			poiService.delete(null);
		});
		assertThrows(POIException.class, () -> {
			POIDTO dto = new POIDTO(null, null, null);
			poiService.delete(dto);
		});
		assertThrows(POIException.class, () -> {
			POIDTO dto = new POIDTO("", null, null);
			poiService.delete(dto);
		});
		// Desabilitado para base em atividade
		/*
		 * assertThrows(POIException.class, () -> { POIDTO dto = new POIDTO("A", null,
		 * null); poiService.delete(dto); }); POIDTO dtoO = new POIDTO("G", null, null);
		 * 
		 * poiService.delete(dtoO);
		 */
	}

	// Desabilitado para não alterar a base de dados
	// Testes em deleteAll
	// @Test
	public void deleteAll() throws POIException {
		poiService.deleteAll();
	}
}

File: src/test/kotlin/br/campos/poi/POIServiceTests.kt
package br.campos.poi

import br.campos.poi.model.transfer.POIDTO
import br.campos.poi.service.abst.IPOIService
import br.campos.poi.util.POIException
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.junit.jupiter.SpringExtension

/**
 * Testes para o componente IPOIService e POIService
 *
 * @author Caio
 */
@AutoConfigureMockMvc
@ExtendWith(SpringExtension::class)
@SpringBootTest(classes = [POIIncApplication::class])
class POIServiceTests {

	@Autowired
	lateinit var poiService: IPOIService

	private val mOne = -1.0
	private val dValue = 1435674.003

	// Testes em create
	@Test
	fun create() {
		assertThrows(POIException::class.java) { poiService.create(null) }
		assertThrows(POIException::class.java) { poiService.create(POIDTO(null, null, null)) }
		assertThrows(POIException::class.java) { poiService.create(POIDTO(null, mOne, null)) }
		assertThrows(POIException::class.java) { poiService.create(POIDTO(null, null, mOne)) }
		assertThrows(POIException::class.java) { poiService.create(POIDTO("A", null, null)) }
		assertThrows(POIException::class.java) { poiService.create(POIDTO("B", mOne, null)) }
		assertThrows(POIException::class.java) { poiService.create(POIDTO("C", null, mOne)) }
		assertThrows(POIException::class.java) { poiService.create(POIDTO("D", mOne, dValue)) }
		assertThrows(POIException::class.java) { poiService.create(POIDTO("E", dValue, mOne)) }
		assertThrows(POIException::class.java) { poiService.create(POIDTO("", dValue, dValue)) }
		// Desabilitado para base em atividade
		// val dtoO = POIDTO("F", dValue, dValue)
		// assertNotNull(poiService.create(dtoO))
		// assertThrows(POIException::class.java) { poiService.create(POIDTO("F", dValue, dValue)) }
		// dtoO = POIDTO("G", 0.0, 10.0)
		// assertNotNull(poiService.create(dtoO))
		// dtoO = POIDTO("H", 10.0, 0.0)
		// assertNotNull(poiService.create(dtoO))
	}

	// Testes em update
	@Test
	fun update() {
		assertThrows(POIException::class.java) { poiService.update(null) }
		assertThrows(POIException::class.java) { poiService.update(POIDTO(null, null, null)) }
		assertThrows(POIException::class.java) { poiService.update(POIDTO("", dValue, dValue)) }
		assertThrows(POIException::class.java) { poiService.update(POIDTO("F", mOne, null)) }
		assertThrows(POIException::class.java) { poiService.update(POIDTO("F", null, mOne)) }
		assertThrows(POIException::class.java) { poiService.update(POIDTO("F", mOne, dValue)) }
		assertThrows(POIException::class.java) { poiService.update(POIDTO("F", dValue, mOne)) }
		// Desabilitado para base em atividade
		// assertThrows(POIException::class.java) { poiService.update(POIDTO("A", dValue, dValue)) }
		// val dtoO = POIDTO("F", 1.0, 10.0)
		// assertNotNull(poiService.update(dtoO))
	}

	// Testes em find e findDTO
	@Test
	fun findDTO() {
		assertThrows(POIException::class.java) { poiService.findDTO(null) }
		assertThrows(POIException::class.java) { poiService.findDTO(POIDTO(null, null, null)) }
		assertThrows(POIException::class.java) { poiService.findDTO(POIDTO("", null, null)) }
		// Desabilitado para base em atividade
		// assertThrows(POIException::class.java) { poiService.findDTO(POIDTO("A", null, null)) }
		// val dtoO = POIDTO("H", null, null)
		// assertNotEquals(poiService.findDTO(dtoO).toString(), "POIDTO(nome=H, x=10, y=0)")
		// assertTrue(poiService.findDTO(0.0, 0.0, 0.0).isEmpty())
		// assertTrue(poiService.findDTO(0.0, 0.0, 10.0).size == 2)
		// assertTrue(poiService.findDTO().size == 3)
	}

	// Testes em delete
	@Test
	fun delete() {
		assertThrows(POIException::class.java) { poiService.delete(null) }
		assertThrows(POIException::class.java) { poiService.delete(POIDTO(null, null, null)) }
		assertThrows(POIException::class.java) { poiService.delete(POIDTO("", null, null)) }
		// Desabilitado para base em atividade
		// assertThrows(POIException::class.java) { poiService.delete(POIDTO("A", null, null)) }
		// val dtoO = POIDTO("G", null, null)
		// poiService.delete(dtoO)
	}

	// Desabilitado para não alterar a base de dados
	// Testes em deleteAll
	// @Test
	fun deleteAll() {
		poiService.deleteAll()
	}
}
